include/micm/util/cuda_vector_matrix.hpp

```diff
@@ -0,0 +1,65 @@
+#include <type_traits>
+#include <micm/util/vector_matrix.hpp>
+#include <micm/util/cuda_vector_matrix.cuh>
+
+namespace micm {
+
+    template<class T, std::size_t L = DEFAULT_VECTOR_SIZE>
+    class CudaVectorMatrix : public VectorMatrix<T, L> {
```

Comment:
I suggest adding a description in doxygen format for at least the class as a whole, describing what it's purpose is, how it allows transfer to/from device memory, etc.

Reply:
Added documentation, just let me know if it could be clearer or needs some formatting updates.